API/Services/BookService.cs

```diff
@@ -425,10 +426,189 @@ private async Task InlineStyles(HtmlDocument doc, EpubBookRef book, string apiBa
+        ns.AddNamespace("prism", "http://prismstandard.org/namespaces/basic/2.0/");
+        ns.AddNamespace("xmp", "http://ns.adobe.com/xap/1.0/");
+
+        var info = new ComicInfo();
+        var publicationDate = GetDateTimeFromXmlNode(metaDoc, ns, "//dc:date")
+            ?? GetDateTimeFromXmlNode(metaDoc, ns, "//xmp:createdate");
+        if (publicationDate != null)
+        {
+            info.Year  = publicationDate.Value.Year;
+            info.Month = publicationDate.Value.Month;
+            info.Day   = publicationDate.Value.Day;
+        }
+        info.Summary   = GetTextFromXmlNode(metaDoc, ns, "//dc:description") ?? "";
+        info.Publisher = GetTextFromXmlNode(metaDoc, ns, "//dc:publisher") ?? "";
+        info.Writer    = GetListFromXmlNode(metaDoc, ns, "//dc:creator") ?? "";
+        info.Title     = GetTextFromXmlNode(metaDoc, ns, "//dc:title") ?? "";
+        info.Genre     = GetListFromXmlNode(metaDoc, ns, "//dc:subject") ?? "";
+        info.LanguageISO = ValidateLanguage(GetTextFromXmlNode(metaDoc, ns, "//dc:language"));
+        ComicInfo.CleanComicInfo(info);
+
+        info.Isbn = GetTextFromXmlNode(metaDoc, ns, "//pdfx:isbn") ?? GetTextFromXmlNode(metaDoc, ns, "//prism:isbn") ?? "";
+        if (!ArticleNumberHelper.IsValidIsbn10(info.Isbn) && !ArticleNumberHelper.IsValidIsbn13(info.Isbn))
+        {
+            _logger.LogDebug("[BookService] {File} has invalid ISBN number", filePath);
+            info.Isbn = "";
+        }
+
+        info.UserRating = GetFloatFromXmlNode(metaDoc, ns, "//calibre:rating") ?? 0.0f;
+        info.TitleSort  = GetTextFromXmlNode(metaDoc, ns, "//calibre:title_sort") ?? "";
+        info.Series     = GetTextFromXmlNode(metaDoc, ns, "//calibre:series/rdf:value") ?? "";
+        info.SeriesSort = info.Series;
+        info.Volume     = Convert.ToInt32(GetFloatFromXmlNode(metaDoc, ns, "//calibreSI:series_index") ?? 0.0f).ToString();
+
+        // If this is a single book and not a collection, set publication status to Completed
+        if (string.IsNullOrEmpty(info.Volume) && Parser.ParseVolume(filePath, LibraryType.Manga).Equals(Parser.LooseLeafVolume))
+        {
+            info.Count = 1;
+        }
+
+        var hasVolumeInSeries = !Parser.ParseVolume(info.Title, LibraryType.Manga)
+            .Equals(Parser.LooseLeafVolume);
+
+        if (string.IsNullOrEmpty(info.Volume) && hasVolumeInSeries && (!info.Series.Equals(info.Title) || string.IsNullOrEmpty(info.Series)))
+        {
+            // This is likely a light novel for which we can set series from parsed title
+            info.Series = Parser.ParseSeries(info.Title, LibraryType.Manga);
+            info.Volume = Parser.ParseVolume(info.Title, LibraryType.Manga);
+        }
+
+        return info;
+    }
+
+    private ComicInfo? GetPdfComicInfo(string filePath)
     {
-        if (!IsValidFile(filePath) || Parser.IsPdf(filePath)) return null;
+        try
+        {
+            const int chunkSize = 4096;
```

Comment:
This is a very technical implementation. Is it not possible to get the comment field from docnet then parse out the xml from that?

Reply:
As far as I could tell from the sparse documentation, Docnet is rather page oriented, and the metadata does not seem to be attached to any particular page. And I don't see APIs to access non-visible parts of the PDF file anyway.

It's not clear to me whether PDFium has support for this metadata (documentation is downright nonexistent). But I see neither support for returning the entire XML blob as a whole, nor specific support for calibre specific info.

So I don't see how to make this approach work.